design/workitems.go

```diff
@@ -112,6 +112,7 @@ var _ = a.Resource("workitem", func() {
 			a.Param("filter[iteration]", d.String, "IterationID to filter work items")
 			a.Param("filter[workitemtype]", d.String, "work item type to filter work items by")
 			a.Param("filter[area]", d.String, "AreaID to filter work items")
+			a.Param("filter[workitemstate]", d.String, "work item state to filter work items by")
```

Comment:
@baijum i think `workitemstate` more apt because firstly it's consistent as the `workitemtype` and secondly state do not clear the context that much. WDYT?
Thanks for the review!